src/element-stub.js

```diff
@@ -15,21 +15,110 @@
  */
 
 import {BaseElement} from './base-element';
-
+import {getMode} from '../src/mode';
 
 /** @type {!Array} */
 export const stubbedElements = [];
 
+/**
+* This value tells the ampAdScript is already presented or inserted.
+* Set the value to true when querySelector find an amp-ad script
+* or when add amp-ad script to head.
+* @type {boolean}
+*/
+let ampAdScriptInsertedOrPresent = false;
+
+/**
+* @visibleForTesting
+* Reset the ampAdScriptInsertedOrPresent value for each test.
+*/
+export function resetAdScriptInsertedOrPresentForTesting() {
+  ampAdScriptInsertedOrPresent = false;
+}
+
 export class ElementStub extends BaseElement {
   constructor(element) {
     super(element);
+    this.updateAmpAdScriptInfo(element);
     stubbedElements.push(this);
   }
 
+  /**
+   * Check script info in HTML head and make update if necessary
+   * @param {!Element} element
+  */
+  updateAmpAdScriptInfo(element) {
```

Comment:
This kind of functionality will be necessary for anybody who wants to lazily load an extension.  (I'd like it for the a4a module (issue #3133) that I'm working on, for example.)  Would it be worth factoring this stuff out to a more general "insert extension X if not already present" set of code in src/?